lessons/module-2/closures.md

```diff
@@ -148,15 +157,120 @@ Our most thorough definition of a closure is now **when an inner function has ac
+
+</section>
+
+## Closures for drying up code
+Consider this next example when looking at the next example: closures can 'partially apply' functions to 'lock in' arguments to make more reusable functions.
+
+```js
+const calculateTaxes = (taxRate) => {
+  return (income) => {
+    return income * taxRate
+  }
+}
+
+const taxRateFor2020 = calculateTaxes(0.24)
+const taxRateFor2021 = calculateTaxes(0.27)
+taxRateFor2020(60000)
+taxRateFor2020(85000)
+taxRateFor2021(95000)
+taxRateFor2022(110000)
+```
+In this code, `calculateTaxes()` is a **higher-order function** that takes a taxRate parameter and returns a new function that takes an income parameter and calculates the tax owed for that income at the given taxRate.
+
+The returned function is a closure, meaning it has access to the taxRate parameter from the outer function even after the outer function has returned. This allows us to "dry up" our code by avoiding duplication of tax calculation logic.
+
+In the subsequent lines of code, we call `calculateTaxes()` twice, once with a taxRate of `0.24` to create a function `taxRateFor2020`, and once with a taxRate of `0.27` to create a function `taxRateFor2021`. We then call each of these functions with different income values to calculate the tax owed.
+
+Using closures in this way allows us to create reusable functions for calculating taxes at different rates, without having to repeat the same logic for each rate. It also allows us to easily calculate taxes for different incomes without having to pass the `taxRate` parameter every time we call the function.
+
 <section class="call-to-action">
 ### Practice
 
-Use the closure pattern to create a "private" counter. The counter variable should be protected by an outter function. The outter function should return an object of methods that allow a user to `incrementCounter`, `decrementCounter` and `getCounterValue`. 
+Consider the following code:
+```js
+function counter() {
+  let count = 0;
+
+  function increment() {
+    count++;
+    console.log(count);
+  }
+
+  increment();
+  increment();
+  increment();
+}
+
+counter(); // Output: 1 2 3
+```
+
+We're using a closure here, which is great! However, when we run our counter function, it only counts to three everytime. Update this function to return a function that allows us to create many counters that keep track of their count separately. 
+
+<section class="answer">
+### Potential Solution
+
+```js
+function createCounter() {
+  let count = 0;
+
+  function increment() {
+    count++;
+    console.log(count);
+  }
+
+  return increment;
+}
+
+const counter = createCounter();
+
+counter(); // Output: 1
+counter(); // Output: 2
+counter(); // Output: 3
+
+const counter2 = createCounter();
+
+counter2(); // Output: 1
+```
+</section>
+***Extra Spicy:*** Include the private methods `incrementCounter`, `decrementCounter` and `getCounterValue` within the closure.
+
 </section>
```

Comment:
It's great that we're showing them examples and giving them opportunities to refactor. Would it be an appropriate challenge to then add an example where they have to build some functions w/ closures from scratch? If that's too spicy, maybe offering one as an optional spicy challenge for students who want that challenge?